ui/app/controllers/topology.js

```diff
@@ -1,23 +1,152 @@
+  @selection('qpVersion') selectionVersion;
+
+  @computed
+  get optionsState() {
+    return [
+      { key: 'initializing', label: 'Initializing' },
+      { key: 'ready', label: 'Ready' },
+      { key: 'down', label: 'Down' },
+      { key: 'ineligible', label: 'Ineligible' },
+      { key: 'draining', label: 'Draining' },
+      { key: 'disconnected', label: 'Disconnected' },
+    ];
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionClass')
+  get optionsClass() {
+    const classes = Array.from(new Set(this.model.nodes.mapBy('nodeClass')))
+      .compact()
+      .without('');
+
+    // Remove any invalid node classes from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpClass',
+        serialize(intersection(classes, this.selectionClass))
+      );
+    });
+
+    return classes.sort().map((dc) => ({ key: dc, label: dc }));
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionDatacenter')
+  get optionsDatacenter() {
+    const datacenters = Array.from(
+      new Set(this.model.nodes.mapBy('datacenter'))
+    ).compact();
+
+    // Remove any invalid datacenters from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpDatacenter',
+        serialize(intersection(datacenters, this.selectionDatacenter))
+      );
+    });
+
+    return datacenters.sort().map((dc) => ({ key: dc, label: dc }));
+  }
+
+  @computed('model.nodes', 'nodes.[]', 'selectionVersion')
+  get optionsVersion() {
+    const versions = Array.from(
+      new Set(this.model.nodes.mapBy('version'))
+    ).compact();
+
+    // Remove any invalid versions from the query param/selection
+    scheduleOnce('actions', () => {
+      // eslint-disable-next-line ember/no-side-effects
+      this.set(
+        'qpVersion',
+        serialize(intersection(versions, this.selectionVersion))
+      );
+    });
+
+    return versions.sort().map((v) => ({ key: v, label: v }));
+  }
+
   @alias('userSettings.showTopoVizPollingNotice') showPollingNotice;
 
-  @tracked filteredNodes = null;
+  @tracked pre09Nodes = null;
+
+  get filteredNodes() {
+    const { nodes } = this.model;
+    return nodes.filter((node) => {
+      const {
+        searchTerm,
+        selectionState,
+        selectionVersion,
+        selectionDatacenter,
+        selectionClass,
+      } = this;
+      return (
+        (selectionState.length ? selectionState.includes(node.status) : true) &&
+        (selectionVersion.length
+          ? selectionVersion.includes(node.version)
+          : true) &&
+        (selectionDatacenter.length
+          ? selectionDatacenter.includes(node.datacenter)
+          : true) &&
+        (selectionClass.length
+          ? selectionClass.includes(node.nodeClass)
+          : true) &&
+        node.name.includes(searchTerm)
```

Comment:
I feel like node name or datacenter might be nice here

Reply:
Sounds good, added for class and datacenter as well.